### PR TITLE
[4.0] Add a filter featured to com_contact

### PR DESCRIPTION
**Pull request by hardik-codes**
### Summary of Changes
Presently, it is not possible to filter for unfeatured contacts in com_contact.
This PR adds a filter "featured" to com_contact

### Testing Instructions
Ensure that you have at least one featured and at least one unfeatured contacts.
Apply the Patch and the list of filters in the Filter Options has a new filter "-Select Featured-".
Apply the filter to your contacts

### Documentation Changes Required
Yes 
New Filter

### CODE REVIEW

File: administrator/components/com_contact/forms/filter_contacts.xml [hunk 1]
```diff
@@ -11,6 +11,17 @@
 			hint="JSEARCH_FILTER"
 		/>
 
+		<field
```
Comment (brianteeman) [inline administrator/components/com_contact/forms/filter_contacts.xml:14]: the fields should be ordered to match the order they appear in the table. so in this case please move this field after the published field

Comment (hardik-codes) [reply administrator/components/com_contact/forms/filter_contacts.xml:14]: Ok I'll do so

File: administrator/components/com_contact/Model/ContactsModel.php [hunk 2]
```diff
@@ -251,6 +251,13 @@ protected function getListQuery()
 				);
 		}
 
+		// Filter by featured.
+		$featured = (string) $this->getState('filter.featured');
+		if (in_array($featured, ['0','1']))
```
Comment (Quy) [inline administrator/components/com_contact/Model/ContactsModel.php:257]:
```suggestion

		if (in_array($featured, ['0','1']))
```

File: administrator/components/com_contact/forms/filter_contacts.xml (hunk 1 above)
Comment (hardik-codes) [reply administrator/components/com_contact/forms/filter_contacts.xml:14]: @brianteeman which table are you referring to?

File: administrator/components/com_contact/Model/ContactsModel.php [hunk 3]
```diff
@@ -251,6 +251,14 @@ protected function getListQuery()
 				);
 		}
 
+		// Filter by featured.
+		$featured = (string) $this->getState('filter.featured');
+
+		if (in_array($featured, ['0','1']))
+		{
+			$query->where('a.featured =' . (int) $featured);
```
Comment (Quy) [inline administrator/components/com_contact/Model/ContactsModel.php]: Change it to be line 265 using `quoteName`.

Comment (hardik-codes) [reply administrator/components/com_contact/Model/ContactsModel.php]: @Quy could you please elaborate?

Comment (alikon) [reply administrator/components/com_contact/Model/ContactsModel.php]: `$query->where('a.featured'`  should be something like
`$query->where($db->quoteName('a.featured')`  if i understand  @Quy correctly

Comment (Quy) [reply administrator/components/com_contact/Model/ContactsModel.php]: Yes as alikon stated.

Comment (hardik-codes) [reply administrator/components/com_contact/Model/ContactsModel.php]: @Quy as stated by you to use ```quoteName``` in line 256, but I think that should go in line 259
as per the suggestion given by @alikon .

Comment (Quy) [reply administrator/components/com_contact/Model/ContactsModel.php]: Line 265 is for reference. The change should be done to line 259.

**Comment by hardik-codes** [comment]
@brianteeman @alikon @Quy  I have updated the files
Please review the PR.

File: administrator/components/com_contact/forms/filter_contacts.xml (hunk 1 above)
Comment (Quy) [reply administrator/components/com_contact/forms/filter_contacts.xml:14]: Move lines 23-32 before line 14.

Comment (hardik-codes) [reply administrator/components/com_contact/forms/filter_contacts.xml:14]: done @Quy

**Comment by pranotiTechjoomla** [comment]
I have tested this item :white_check_mark: successfully on d1293437c825347c798c39428bf01d5c2d1da40d<br /><br />I have successfully test issue<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24167">issues.joomla.org/tracker/joomla-cms/24167</a>.</sub>

**Comment by pravinTek** [comment]
I have tested this item :white_check_mark: successfully on d1293437c825347c798c39428bf01d5c2d1da40d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24167">issues.joomla.org/tracker/joomla-cms/24167</a>.</sub>

**Comment by nidhi1709** [comment]
I have tested this item :white_check_mark: successfully on d1293437c825347c798c39428bf01d5c2d1da40d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24167">issues.joomla.org/tracker/joomla-cms/24167</a>.</sub>

**Comment by HLeithner** [comment]
thx

**Comment by neotrendch** [comment]
I have tested this item :white_check_mark: successfully on d1293437c825347c798c39428bf01d5c2d1da40d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24167">issues.joomla.org/tracker/joomla-cms/24167</a>.</sub>

**Pull request merged**